bridge/tracker/rootchain.go

```diff
@@ -0,0 +1,236 @@
+package tracker
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/0xPolygon/polygon-edge/blockchain/storage"
+	"github.com/0xPolygon/polygon-edge/blockchain/storage/leveldb"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+	"github.com/umbracle/go-web3"
+	"github.com/umbracle/go-web3/abi"
+	client "github.com/umbracle/go-web3/jsonrpc"
+	"math/big"
+	"os"
+	"path/filepath"
+)
+
+const (
+	//	required block depth for fetching events on the rootchain
+	DefaultBlockConfirmations = 6
+
+	//	Ropsten testnet
+	//rootchainWS   = "wss://ropsten.infura.io/ws/v3/17eac086ff36442ebd43737400eb71ca"
+
+	//	Polygon Edge
+	rootchainWS = "ws://127.0.0.1:10002/ws"
+)
+
+//	contractABI is used to create query filter
+//	that matches events defined in a smart contract.
+type contractABI struct {
+	//	address of smart contract
+	address web3.Address
+
+	// signatures of events defined in smart contract
+	events []*abi.Event
+}
+
+//	eventIDs returns all the event signatures (IDs)
+//	defined in the smart contract.
+func (c *contractABI) eventIDs() (IDs []*web3.Hash) {
+	for _, ev := range c.events {
+		id := ev.ID()
+		IDs = append(IDs, &id)
+	}
+
+	return
+}
+
+//	loadABIs parses contracts from raw map.
+func loadABIs(abisRaw map[string][]string) (contracts []*contractABI) {
```

Comment:
Either's fine. But map key is better to be types.Address?

Reply:
[Resolved](https://github.com/0xPolygon/polygon-edge/pull/386/commits/585d357ff9ff834ac33cc2fd98a060539046f5dc)